### DISCUSSION TITLE
Connection timeout and other

### DISCUSSION THREAD

**Discussion by morkioj** [philibertc/micronova_controller#13] [General]
Hello philibert, 
first of all congratulations on your project. Is fantastic. I bought the pcb on tindie.
I have any questions for you:
1. If my esp starts without network (eg. blackout), my esp after n attemps switch to ap mode because the wifi is not yet available. how can i solve?
2.  Do you want add this lines after _ambTemp= val /2_  in checkStoveReply method (case ambTempAddr:)? This instruction get the 0.5 °C
3. The address 3E of tempFumes always returns 10. The address 5A returns 0. I tested randomly 3A and this returns 80 (0x50). Is this?

My stove is Anselmo Cola (vendor) Aloe (model). The micronova model is O100
thank you so much

**Comment by philibertc** [comment]
Hello! Glad you like this project! 

1. I believe that if you restart the ESP after your WiFi network has returned it will reconnect normally
2. Add which line? Sorry, I didn't understand your question
3. You can check if this value corresponds to what is shown on the display of the stove. I am making a test program that reads all the addresses so we will soon be able to find out which one is the right one.

**Reply by morkioj** [reply]
1. OK!
2. Sorry, I forgot the lines in the post. This lines (to get 0.5 values of temperature):
`            if((val % 2)>0)
            {
              ambTemp = ambTemp + 0.5;
            }`
3. effectively, 3A doesn't work. IT returns every 80, even stove off

**Reply by philibertc** [reply]
In fact the val / 2 works for the .5
But it was an uint8 and not a float so the .5 was getting deleted
Now corrected!

**Reply by philibertc** [reply]
Hello,
In fact, I noticed that the .5 still does not appear. I'm working on it!

**Reply by philibertc** [reply]
Solved, this time!

**Comment by philibertc** [comment]
Hello, is everything working properly now?
